src/AWS/CRT/HTTP/Message.php

```diff
@@ -2,32 +2,94 @@
 
 namespace AWS\CRT\HTTP;
 
-abstract class Message {
+use AWS\CRT\NativeResource;
+
+use AWS\CRT\Internal\Encoding;
+
+abstract class Message extends NativeResource {
+    private $method = "GET";
```

Comment:
Why you setting it to GET here if it is overrode in the contructor? 
Same for other private variables.

Reply:
I have a pathological hatred of uninitialized variables. In this case, it makes sense to just remove all the values, which I've done :(